docs/gameplay-matchmaker.md

```diff
@@ -1,182 +1,351 @@
+```csharp fct_label=".NET"
+var query = "*"
+var minCount = 4;
+var maxCount = 4;
+
+var matchmakerTicket = await socket.AddMatchmakerAsync(query, minCount, maxCount);
 ```
 
 ```csharp fct_label="Unity"
-// Updated example TBD
+var query = "*"
+var minCount = 4;
+var maxCount = 4;
+
+var matchmakerTicket = await socket.AddMatchmakerAsync(query, minCount, maxCount);
 ```
 
-### Multi-filter matchmaking
+## Matchmaker tickets
 
-You can of course mix and match various filters to enhance the matchmaking search.
+Each time a user is added to the matchmaker pool they receive a ticket, a unique identifier for their entry into the pool.
 
 ```js fct_label="JavaScript"
-// Updated example TBD
+const message = { matchmaker_add: {
+  min_count: 2,
+  max_count: 4,
+  query: "*"
+}};
+var ticket = await socket.send(message);
 ```
 
-```csharp fct_label=".Net"
-// Updated example TBD
+```csharp fct_label=".NET"
+var query = "*"
+var minCount = 2;
+var maxCount = 4;
+
+var matchmakerTicket = await socket.AddMatchmakerAsync(query, minCount, maxCount);
 ```
 
 ```csharp fct_label="Unity"
-// Updated example TBD
+var query = "*"
+var minCount = 2;
+var maxCount = 4;
+
+var matchmakerTicket = await socket.AddMatchmakerAsync(query, minCount, maxCount);
 ```
 
-## Cancel a request
+This ticket is used to notify the client when matching completes to distinguish between multiple matchmaker operations. User may also cancel the matchmaking process using the ticket.
 
-After a user has sent a message to add themselves to the matchmaker pool you'll receive a ticket which can be used to cancel the action at some later point.
+## Remove a user from the matchmaker
 
-Users may cancel matchmake actions at any point after they've added themselves to the pool but before they've been matched. Once matching completes for the ticket it can no longer be used to cancel.
+If a user decides they no longer wish to matchmake without disconnecting they can gracefully cancel the matchmaker process by removing themselves from the pool.
 
 ```js fct_label="JavaScript"
-const message = { matchmaker_remove: {
-  ticket: "<matchmakerticket>"
-}};
+// `ticket` is returned by matchmaker add operations.
+const message = {
+  matchmaker_remove: {
+    ticket: ticket
+  }
+}
 socket.send(message);
 ```
 
-```csharp fct_label=".Net"
-// Updated example TBD
+```csharp fct_label=".NET"
+// `matchmakerTicket` is returned by matchmaker add operations.
+socket.RemoveMatchmakerAsync(matchmakerTicket);
 ```
 
 ```csharp fct_label="Unity"
-// Updated example TBD
+// `matchmakerTicket` is returned by matchmaker add operations.
+socket.RemoveMatchmakerAsync(matchmakerTicket);
 ```
 
-The user is now removed from the matchmaker pool.
+If the user has multiple entries in the matchmaker only the one identified by the ticket will be removed.
 
-!!! Note "Removed on disconnect"
-    A user will automatically be removed from the matchmaker pool by the server when they disconnect.
+## Receive matchmaker results
```

Comment:
Let's talk about MatchMakerMatched function in Lua as well here - how does someone add custom logic after matchmaking is complete?

Reply:
Maybe as a separate section